### PR TITLE
Adding AzDO pipeline with matrix build and code coverage

### LINKED ISSUE #614: Create an issue for CI/CD ADO for SPK

### PR DESCRIPTION
**Pull request by andrebriggs**
Adding an azure-pipelines.yaml and configuring jest for code coverage.

- Example of AzDO pipeline here: https://dev.azure.com/abrig/spk/_build/results?buildId=475&view=results
- Example of code coverage here: https://dev.azure.com/abrig/spk/_build/results?buildId=475&view=codecoverage-tab

Next Steps after merge:
- Import the azure-pipeline.yml file to an "official" SPK AzDO project

Closes https://github.com/microsoft/bedrock/issues/614

### CODE REVIEW

**Comment by mydiemho** [comment]
thoughts on also running this on macos and windows?

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+- script: |
+    yarn test
+  displayName: 'YARN TEST'
+
+- task: PublishCodeCoverageResults@1
+  inputs: 
+    codeCoverageTool: Cobertura
+    summaryFileLocation: '$(System.DefaultWorkingDirectory)/**/*coverage.xml'
+    reportDirectory: '$(System.DefaultWorkingDirectory)/**/coverage'
```
Comment (evanlouie) [inline azure-pipelines.yml]: new line please 🙏

**Review by evanlouie — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -20,6 +20,7 @@
     "husky": ">=1",
     "jest": "^24.9.0",
     "lint-staged": ">=8",
+    "nyc": "^14.1.1",
```
Comment (evanlouie) [inline package.json:23]: What are we using nyc for?

Comment (andrebriggs) [reply package.json:23]: Needed for coverage

**Review by bnookala — APPROVED** [review]

**Pull request merged**